log-viewer/src/main/java/com/logviewer/web/session/LogSession.java

```diff
@@ -57,7 +62,7 @@
     @Autowired(required = false)
     private List<LvPathResolver> pathResolvers = Collections.emptyList();
 
-    private final List<SessionTask<?>> executions = new LinkedList<>();
+    private final Queue<SessionTask<?>> executions = new ConcurrentLinkedQueue<>();
```

Comment:
Let's keep the contract: "all fields in LogSession are accessible under synchronize(LogSession.this)". Mixing synchronization and lock-free algorithms may be too complex